test/cmocka/src/audio/selector/selector_test.c

```diff
@@ -198,11 +330,20 @@ static void fill_source_s32(struct sel_test_state *sel_state)
 	audio_stream_produce(stream, audio_stream_get_free_bytes(stream));
 }
 
+#if CONFIG_IPC_MAJOR_3
```

Comment:
not sure I understand, I thought switching to `struct processing_module` was a part of the module adapter support, is it really related to IPC3 vs. IPC4? Isn't the module adapter API used with IPC3 too?

Reply:
There is a issue : https://github.com/thesofproject/sof/pull/5973#issuecomment-1206062342. And Liam suggested using IPC to choose it like volume

Reply:
to be simple: dai is for ipc3 but doesn't support module adapter and there is no dai for ipc4. selector needs to match up with dai.